include/seqan3/alignment/pairwise/edit_distance_unbanded.hpp

```diff
@@ -495,6 +495,54 @@ struct edit_distance_unbanded_trace_matrix_policy :
     //!\}
 };
 
+//!\brief The same as `T &` but it is default constructible and is re-assignable.
+template <typename T>
+struct proxy_reference
+{
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    proxy_reference() noexcept = default;                                    //!< Defaulted.
+    proxy_reference(proxy_reference const &) noexcept = default;             //!< Defaulted.
+    proxy_reference(proxy_reference &&) noexcept = default;                  //!< Defaulted.
+    proxy_reference & operator=(proxy_reference const &) noexcept = default; //!< Defaulted.
+    proxy_reference & operator=(proxy_reference &&) noexcept = default;      //!< Defaulted.
+    ~proxy_reference() = default;                                            //!< Defaulted.
+
+    //!\brief Use the lvalue `t` as the stored reference.
+    proxy_reference(T & t) noexcept
+        : ptr(std::addressof(t))
+    {}
+
+    proxy_reference(T &&) = delete; //!< Deleted.
+
+    //!\brief Assign a value to the stored reference.
+    template <typename U>
+    proxy_reference & operator=(U && u) noexcept
```

Comment:
This should be constrained. ConvertibleTo<U, T> minus the references. Also please don‘t use Capital letters for the types. We don‘t do this anywhere else. I know it is tempting because cppref but we should keep it consistent